atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlQueryContext.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async;
+
+import java.util.concurrent.Executor;
+
+import org.immutables.value.Value;
+
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+@Value.Immutable
+public interface CqlQueryContext {
+    String keySpace();
+
+    TableReference tableReference();
+
+    @Value.Auxiliary
+    Executor executor();
```

Comment:
Why does the query context know about the executor?

Reply:
The way I imagined it is that the `CqlClientImpl` will be responsible just for executing the prepared queries on correct session. While the `AsyncCellLoader` (or some other class for future queries) will know what to do with results (how to combine the results and on which pool, etc.).